src/forms/TimeInput.jsx

```diff
@@ -58,8 +58,9 @@ class TimeInput extends React.Component {
 	 */
 	parseValueIntoState(value, is24Hr) {
 		const timeParts = getTimeParts(value);
+
 		return {
-			hours: value && (formatDigits(timeParts.hours % (is24Hr ? 24 : 12)) || '12'),
+			hours: value && (formatDigits(timeParts.hours % (is24Hr ? 24 : 12) || (is24Hr ? '00' : '12'))), // TODO test midnight at is24
```

Comment:
this is hard to read - probably should be setting more named `const`s before assembling them, but not a blocker

Reply:
Agreed.